app.py

```diff
@@ -935,18 +981,21 @@ def export_gpx():
             gpx_segment = gpxpy.gpx.GPXTrackSegment()
             gpx_track.segments.append(gpx_segment)
             gps_data = (
-                json.loads(trip["gps"]) if isinstance(trip["gps"], str) else trip["gps"]
+                json.loads(trip["gps"]) if isinstance(
+                    trip["gps"], str) else trip["gps"]
             )
             if gps_data.get("type") == "LineString":
                 for coord in gps_data.get("coordinates", []):
                     if isinstance(coord, list) and len(coord) >= 2:
                         lon, lat = coord[0], coord[1]
-                        gpx_segment.points.append(gpxpy.gpx.GPXTrackPoint(lat, lon))
+                        gpx_segment.points.append(
+                            gpxpy.gpx.GPXTrackPoint(lat, lon))
             elif gps_data.get("type") == "Point":
                 coord = gps_data.get("coordinates", [])
                 if isinstance(coord, list) and len(coord) >= 2:
                     lon, lat = coord[0], coord[1]
-                    gpx_segment.points.append(gpxpy.gpx.GPXTrackPoint(lat, lon))
+                    gpx_segment.points.append(
+                        gpxpy.gpx.GPXTrackPoint(lat, lon))
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider consolidating duplicate GPX point logic</b></div>
<div id="fix">

Consider consolidating the duplicate GPX point creation logic into a helper function to reduce code duplication. The same point creation logic appears in two places.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -944,7 +944,11 @@
  def add_gpx_point(segment, lat, lon):
      segment.points.append(gpxpy.gpx.GPXTrackPoint(lat, lon))

  if isinstance(coord, list) and len(coord) >= 2:
      lon, lat = coord[0], coord[1]
 -    gpx_segment.points.append(gpxpy.gpx.GPXTrackPoint(lat, lon))
 +    add_gpx_point(gpx_segment, lat, lon)

  if isinstance(coord, list) and len(coord) >= 2:
      lon, lat = coord[0], coord[1]
 -    gpx_segment.points.append(gpxpy.gpx.GPXTrackPoint(lat, lon))
 +    add_gpx_point(gpx_segment, lat, lon)
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/realronaldrump/everystreet-new/pull/98#issuecomment-2549721358>#1b42fb</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged